In [114]:
import numpy as np
import pandas_profiling as pp
import pandas as pd
import plotly.express as px
import scipy.stats as st
import statsmodels.api as ss
import cufflinks as cf
cf.go_offline()

In [31]:
boston_url = 'https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-ST0151EN-SkillsNetwork/labs/boston_housing.csv'
boston_df=pd.read_csv(boston_url)

In [37]:
px.box(boston_df, x='MEDV', title='Description of owner-occupied statistics values',
       labels={'MEDV': 'Owner Occupied'})

In [73]:
px.histogram(boston_df, x='CHAS', color='CHAS', labels={'CHAS':'Close to Charles river'}, 
            title=' Histogram of the Charles river variable')

In [95]:
age_groups = pd.DataFrame(columns=['AGE', 'MEDV'])

age_young= boston_df[boston_df.AGE <= 35][['AGE','MEDV']]
age_young['Type']='young'

age_mid = boston_df[boston_df.eval('AGE > 35 & AGE < 70')][['AGE','MEDV']]
age_mid['Type']='middle'

age_old = boston_df[boston_df.AGE <= 70][['AGE','MEDV']]
age_old['Type']='old'

age_groups = pd.concat([age_young, age_mid, age_old])

px.box(age_groups, y=['MEDV','AGE'], color='Type',
       title='Description of age and median value of owner-occupied statistics values')

In [96]:
px.scatter(boston_df, x='NOX', y='INDUS')

from the above plot we can see that there is a strong positive correlation between them.

In [98]:
px.histogram(boston_df, x='PTRATIO', labels={'PTRATIO': 'Pupil to teacher ratio'},
             title='The pupil to teacher ratio')

In [107]:
st.ttest_ind(boston_df[boston_df['CHAS'] == 1]['MEDV'],boston_df[boston_df['CHAS'] == 0]['MEDV'],
             equal_var = True)

Ttest_indResult(statistic=3.996437466090509, pvalue=7.390623170519905e-05)

H0: Is there a significant difference in median value of houses bounded by the Charles river or not? 
the pvalue is less then 0.05; so we reject the null hypothesis (H0).<br>
there is no significant evidence that at least one of the means differ.

In [110]:
st.levene(boston_df[boston_df.AGE <= 35]['MEDV'],
          boston_df[boston_df.eval('AGE > 35 & AGE < 70')]['MEDV'],
          boston_df[boston_df.AGE <= 70]['MEDV'], center='mean')

LeveneResult(statistic=0.7217477129754917, pvalue=0.4864832347392163)

H0:Is there a difference in Median values of houses (MEDV) for each proportion of owner occupied units built prior to 1940 (AGE)?
the pvalue is > then 0.05; so we fail to reject the null hypothesis (H0).

In [111]:
st.pearsonr(boston_df['NOX'], boston_df['INDUS'])

(0.7636514469209151, 7.913361061238693e-98)

H0: Can we conclude that there is no relationship between Nitric oxide concentrations and proportion of non-retail business acres per town?
the pvalue is < then 0.05; so we reject the null hypothesis (H0).<br>
there is a relationship

In [116]:
X = ss.add_constant(boston_df.DIS) 

model = ss.OLS(boston_df.MEDV, X).fit()
predictions = model.predict(X)

model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   MEDV   R-squared:                       0.062
Model:                            OLS   Adj. R-squared:                  0.061
Method:                 Least Squares   F-statistic:                     33.58
Date:                Wed, 12 May 2021   Prob (F-statistic):           1.21e-08
Time:                        17:45:12   Log-Likelihood:                -1823.9
No. Observations:                 506   AIC:                             3652.
Df Residuals:                     504   BIC:                             3660.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         18.3901      0.817     22.499      0.000      16.784      19.996
DIS            1.0916      0.188      5.795      0.000       0.722       1.462
==============================================================================
Omnibus:                      139.779   Durbin-Watson:                   0.570
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              305.104
Skew:                           1.466   Prob(JB):                     5.59e-67
Kurtosis:                       5.424   Cond. No.                         9.32
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

H0: What is the impact of an additional weighted distance to the five Boston employment centres on the median value of owner occupied homes? the pvalue is < then 0.05; so we reject the null hypothesis (H0).<br>
the impact is the coef 1.0916 so more distance the more the median value.


In [122]:
!ls

 0047416-200613084148143.zip
 0047418-200613084148143.zip
 0.rar
 1.jpeg
 1-s2.0-S1568494620304439-main.pdf
 1zpmJpAxTE-6ZiaQMbxPZg_2a3d9542a1b14356ba9afe348864a309_machine-learning-ex
 1zpmJpAxTE-6ZiaQMbxPZg_2a3d9542a1b14356ba9afe348864a309_machine-learning-ex.zip
'2-ab pruning.mp4'
 2.jpeg
 33586-aws67e99.zip
 3.jpeg
 3_.jpeg
 4.5_Dash_Basics.ipynb
 5_Peer_Graded_Assignment_Questions.ipynb
 747ef0f477cc6b3aa8d722279a4598ca.png
 8542.eps
 8542.svg
 8c97ed8bb0153d953377898328ab375a-fef4b027bbbf2decf875a98b892fb1f34db765df
 8c97ed8bb0153d953377898328ab375a-fef4b027bbbf2decf875a98b892fb1f34db765df.zip
 93250.eps
 abTreePractice-master.zip
 AC-test.MP4
'Action RPG Resources'
'Action RPG Resources.zip'
 Activities.pdf
 Advanced-Machine-Learning-EXAM.docx
'Al-Aodat(1).pdf'
 Al-Aodat.pdf
 AlphaBetaPruningAnimation-master.zip
 amazon-web-services-aws-vector-logo-2208E5627D-seeklogo.com.zip
 AML
 An_Automated_Deep_Learning_Approach_for_Bacterial_.pdf
'[Anime4up.com] KSWKTTNRZ EP 05 HD.mp4'
'an